### ISSUE TITLE
Missing benchmark for `sparse24_sm90_sparsify` overhead

### ISSUE THREAD

**Issue by namgyu-youn** [pytorch/ao#2612]
Hi, I have a question about your team's workshop, [Accelerating Transformer Inference and Training with 2:4 Activation Sparsity](https://arxiv.org/abs/2503.16672). It seems that the benchmark for overhead when converting Squared-ReLU into SparseGEMM is missing.

For the background, Squared-ReLU has high (>85%) sparsity and accuracy loss compared to SWiGLU. Since Squared-ReLU has a high sparsity, it opens a ticket for **compression** to Sparse-GEMM with 2:4 sparsity. Compression logic is in Figure 3 and the following:

https://github.com/pytorch/ao/blob/ebfe1736c4442970835b6eda833c0bc5a1ce2dda/torchao/prototype/sparsity/activation/srelu_linear.py#L57-L64

But I haven't found the overhead for compression (`sparse24_sm90_sparsify`). Does the new benchmark for `sparse24_sm90_sparsify` make sense? Maybe it would be integrated into `benchmarks/benchmark_e2e_fp8_sparse_linear.py`.

**Comment by jainapurva** [comment]
@jcaip Can you please look into this

**Comment by namgyu-youn** [comment]
@jainapurva Sorry for the distraction, but could you let me know who is engaging in the `sparsity` module? Not certain if there is any progress or roadmap because the last commit log was 3 months ago. Even more, there is no comment about sparsity issues (in #2612 and #2517).

Personally, I am looking at other issues but am more interested in `sparsity` module. Could you lead me for deeper contribution?

**Comment by jainapurva** [comment]
@namgyu-youn As of today, we're not actively building sparsity, but we'll support the existing sparsity module. If you're interested in contributing in sparsity, @jcaip is the main POC for it. He's currently on vacation, and I'll be happy to assist in his absence.

**Comment by namgyu-youn** [comment]
> [@namgyu-youn](https://github.com/namgyu-youn) As of today, we're not actively building sparsity, but we'll support the existing sparsity module. If you're interested in contributing in sparsity, [@jcaip](https://github.com/jcaip) is the main POC for it. He's currently on vacation, and I'll be happy to assist in his absence.

@jainapurva Huge thanks for the clear summary; I hope more tickets will be opened for the sparsity module. Personally, I am looking at Activation Compression ( https://github.com/pytorch/ao/issues/1920 ). For example, Vocabulary Trimming prunes dead tokens (embedding compression) specified to SLM (like Qwen3-0.6B and Gemma-3-1b-it). See [here](https://blog.squeezebits.com/vocabulary-trimming-methods#embedding_compression_via_vocabulary_trimming_1) for the reference.

Again, thanks for the clear info. And I will gratefully wait @jcaip !

p.s. Could you please check https://github.com/pytorch/ao/pull/2538 ? It updates annotations in Wanda for better readability. (no code update)

**Comment by jcaip** [comment]
Hi @namgyu-youn I am the only person who works on sparsity in torchao and was on leave the last couple of months, so that's why there hasn't been much activity. 

As for the reason there isn't a `sparsify` standalone benchmark, it's because it's not that interesting - the only thing we would have to compare to would be against the native cuSPARSELt routines, and we know we're faster than those by an order of magnitude. That being said if you added a benchmark I wouldn't mind reviewing it. 

For real world speedups, what we really care about is `sparsify time + sparse gemm time < dense gemm time`, which is measured in the e2e speedups.

**Comment by namgyu-youn** [comment]
> As for the reason there isn't a `sparsify` standalone benchmark, it's because it's not that interesting - the only thing we would have to compare to would be against the native cuSPARSELt routines, and we know we're faster than those by an order of magnitude. That being said if you added a benchmark I wouldn't mind reviewing it.

If I understood correctly, there's currently no benchmark for step 1) (sparsify operation; `sparsify time`), although benchmarks for Sparse-GEMM (`sparse gemm time`) already exist. What I'm suggesting is adding a benchmark for the sparsify step (`sparsify time`) to the existing pipeline.

**Comment by namgyu-youn** [comment]
@jcaip gentle ping for a reply

**Comment by jcaip** [comment]
@namgyu-youn Hey, sorry about missing this earlier. You can compare against `torch._cslt_compress` to get a baseline against the cuSPASRELt provided compression routines. Adding to ` benchmarks/benchmark_e2e_fp8_sparse_linear.py` should be fine.
